core/src/Cuda/Kokkos_Cuda_Parallel_MDRange.hpp

```diff
@@ -253,24 +242,22 @@ class ParallelReduce<FunctorType, Kokkos::MDRangePolicy<Traits...>, ReducerType,
   inline __device__ void exec_range(reference_type update) const {
     Kokkos::Impl::Reduce::DeviceIterateTile<Policy::rank, Policy, FunctorType,
                                             typename Policy::work_tag,
-                                            reference_type>(m_policy, m_functor,
-                                                            update)
+                                            reference_type>(
+        m_policy, m_functor_reducer.get_functor(), update)
```

Comment:
we are storing the functor now twice aren't we? I guess we should store the functor_reducer inside the Iterate class thingy?

Reply:
`DeviceIterateTile` stores a reference, not a copy.

Reply:
@crtrott Do you agree or did I miss something?

Reply:
For the `Serial` implementation we use `HostIterateTile` which stores a copy indeed but in that implementation, we don't have a `m_functor_reducer` member but only a `m_reducer` and a `m_iter` member (the latter one stores the functor).

Reply:
Storing `CombinedFunctorReducer` would be quite disruptive since we are using the same class in `ParallelFor`. Hence, I would prefer just keeping the implementation as is. I would expect that stopping an extra copy of the functor if no reducer is given doesn't make much of a difference on the host anyway.

Reply:
I would be happier to do this if/when we provide `CombinedFunctorReducer` with a call operator.

Reply:
We can do
```C++
diff --git a/core/src/Serial/Kokkos_Serial_Parallel_MDRange.hpp b/core/src/Serial/Kokkos_Serial_Parallel_MDRange.hpp
index 0d7143706..c97880483 100644
--- a/core/src/Serial/Kokkos_Serial_Parallel_MDRange.hpp
+++ b/core/src/Serial/Kokkos_Serial_Parallel_MDRange.hpp
@@ -73,9 +73,8 @@ class ParallelReduce<CombinedFunctorReducerType,
   using value_type     = typename ReducerType::value_type;
   using reference_type = typename ReducerType::reference_type;
 
-  using iterate_type =
-      typename Kokkos::Impl::HostIterateTile<MDRangePolicy, FunctorType,
-                                             WorkTag, reference_type>;
+  using iterate_type = typename Kokkos::Impl::HostIterateTile<
+      MDRangePolicy, CombinedFunctorReducerType, WorkTag, reference_type>;
   const iterate_type m_iter;
   const ReducerType m_reducer;
   const pointer_type m_result_ptr;
@@ -129,7 +128,7 @@ class ParallelReduce<CombinedFunctorReducerType,
   ParallelReduce(const CombinedFunctorReducerType& arg_functor_reducer,
                  const MDRangePolicy& arg_policy,
                  const ViewType& arg_result_view)
-      : m_iter(arg_policy, arg_functor_reducer.get_functor()),
+      : m_iter(arg_policy, arg_functor_reducer),
         m_reducer(arg_functor_reducer.get_reducer()),
         m_result_ptr(arg_result_view.data()) {
     static_assert(Kokkos::is_view<ViewType>::value,
diff --git a/core/src/impl/KokkosExp_Host_IterateTile.hpp b/core/src/impl/KokkosExp_Host_IterateTile.hpp
index 82604a24c..a44ffefa6 100644
--- a/core/src/impl/KokkosExp_Host_IterateTile.hpp
+++ b/core/src/impl/KokkosExp_Host_IterateTile.hpp
@@ -2093,8 +2093,8 @@ struct HostIterateTile<RP, Functor, Tag, ValueType,
     const bool full_tile = check_iteration_bounds(m_tiledims, m_offset);
 
     Tile_Loop_Type<RP::rank, (RP::inner_direction == Iterate::Left), index_type,
-                   Tag>::apply(val, m_func, full_tile, m_offset, m_rp.m_tile,
-                               m_tiledims);
+                   Tag>::apply(val, m_func.get_functor(), full_tile, m_offset,
+                               m_rp.m_tile, m_tiledims);
   }
 
 #else
```
but only if we convert all host backends simultaneously. Thus, I would prefer not doing it here but later.